extension/src/background/background.js

```diff
@@ -31,3 +31,25 @@ function saveWiseSessionToken(token) {
     console.log("Wise session token saved");
   });
 }
+
+chrome.webRequest.onHeadersReceived.addListener(
+  function(details) {
+    // Check if the response has a 401 status code
+    const isUnauthorized = details.statusCode === 401;
+
+    if (isUnauthorized) {
+      // Trigger a message to your content script to handle the logout
+      chrome.tabs.query({ active: true, currentWindow: true }, function(tabs) {
+        chrome.tabs.sendMessage(tabs[0].id, { action: "logout" });
+      });
+    }
+
+    // Continue with the original response
+    return { responseHeaders: details.responseHeaders };
+  },
+  {
+    urls: ["<all_urls>"],
+    types: ["main_frame"],
+  },
+  ["responseHeaders"]
+);
```

Comment:
This checks for unauthorized requests coming through, although it might actually block ANY unauthorized requests so that is no good...